app/services/commits_runner.rb

```diff
@@ -0,0 +1,25 @@
+module CommitsRunner
+  module_function
+
+  def run(commits)
+    commits.each do |commit|
+      if(valid?(commit))
+        create(commit)
+        BenchmarkPool.enqueue(commit[:repo][:name], commit[:sha])
+      end
+    end
+  end
+
+  def valid?(commit)
+    !Commit.merge_or_skip_ci?(commit[:message]) && Commit.valid_author?(commit[:author_name])
+  end
+
+  def create(commit)
+    Commit.find_or_create_by(sha1: commit[:sha]) do |c|
```

Comment:
I think we might need to check if the creation is successfully. Otherwise, it might silently fail.